tests/integrationv2/providers.py

```diff
@@ -308,8 +311,95 @@ def setup_server(self):
         return cmd_line
 
 
-class OpenSSL(Provider):
+class CriterionS2N(S2N):
+    """
+    Wrap the S2N provider in criterion-rust
+    """
+    criterion_off = 'off'
+    criterion_delta = 'delta'
+    criterion_baseline = 'baseline'
+    # Figure out what mode to run in: baseline or delta
+    criterion_mode = get_flag(S2N_USE_CRITERION)
+
+    def _find_s2n_benchmark(self, pattern):
+        # Use executable bit to find the correct file.
+        result = find_files(pattern, root_dir=self.cargo_root, modes=['0o775','0o755'])
+        if len(result) != 1:
+            raise FileNotFoundError(
+                f"Exactly one s2n criterion benchmark not found {result} with expected file mask.")
```

Comment:
This would print out like "Exactly one s2n criterion benchmark not found 0 with expected file mask." Maybe:
```suggestion
            raise FileNotFoundError(
                f"Exactly one s2n criterion benchmark not found. Found {result}.")
```